### PR TITLE
Enforce full View→Compose migration with API audit

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary
- **Phase 8: View API Audit** — mandatory step after Static Verification, blocks device testing until all View API usages are eliminated or user-approved
- **Custom View migration is now mandatory scope** — project-owned custom Views must be fully migrated to Compose composables, `AndroidView` wrapping only allowed for third-party Views (MapView, WebView, PlayerView)
- **New reference: `custom-view-migration.md`** — structured decision tree for migrating custom Views:
  - Classify: composite / behavioral / drawing / hybrid
  - Search for replacements: Material3 → project UI kit → imported libraries → new library
  - Assess custom implementation feasibility per View type
  - Choose strategy: inline / pre-migration component / dedicated sub-task
- Comprehensive prohibited View API list (imports, class refs, method calls, XML references)
- Explicit allowed exceptions with third-party-only `AndroidView` constraint
- Project allowlist support for pre-approved interop usages
- Audit results appended to `migration-report.md`

## Changed files
- `SKILL.md` — new Phase 8 in workflow, custom View mandate in Phase 3
- `references/migration-and-verify.md` — View API Audit scan procedure, prohibited list, allowlist format, report template
- `references/discovery-and-patterns.md` — updated gap analysis and Phase 5 for mandatory custom View migration
- `references/custom-view-migration.md` — new file, full decision tree

## Test plan
- [ ] CI passes
- [ ] Review prohibited API patterns for completeness
- [ ] Review custom View decision tree covers edge cases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md [hunk 1]
```diff
@@ -183,6 +183,101 @@ Brief the `manual-tester` agent with:
 
 The agent captures before/after screenshots, executes all test cases, compares layout/typography/colors/spacing, and populates the screenshot table in the report.
 
+## Phase 8: View API Audit
+
+**This phase is mandatory and blocks device testing.** Its purpose is to guarantee that no deprecated View-system API leaked into the new Compose code.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md:179]: Fixed in aaefd2e. Phase 8 now appears before the Device Testing section.

File: plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md [hunk 2]
```diff
@@ -183,6 +183,101 @@ Brief the `manual-tester` agent with:
...
+
+### Scan procedure
+
+Search all files created or modified during migration (`.kt` files only) for the prohibited patterns below. Use both import scanning and in-body usage scanning — an API can be used via fully qualified name without an import.
+
+### Prohibited API patterns
+
+Scan for any of these in migrated files:
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md]: Fixed in aaefd2e. Added an explicit scope clarification: the audit targets only newly migrated Compose-only files. Legacy host Fragment/Activity files that were only lightly touched during migration are excluded unless being fully rewritten to Compose.

File: plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md [hunk 3]
```diff
@@ -183,6 +183,101 @@ Brief the `manual-tester` agent with:
...
+
+**Imports**
+- `import android.view.*`
+- `import android.widget.*`
+- `import android.app.Activity` (as base class)
+- `import android.app.Fragment` / `import androidx.fragment.app.Fragment`
+- `import android.databinding.*` / `import androidx.databinding.*`
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md:196]: Fixed in aaefd2e. Both Fragment and Activity import entries now carry a heuristic note — import grep is a first-pass signal; the agent must confirm actual supertype usage before flagging.

File: plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md [hunk 4]
```diff
@@ -183,6 +183,101 @@ Brief the `manual-tester` agent with:
 
 The agent captures before/after screenshots, executes all test cases, compares layout/typography/colors/spacing, and populates the screenshot table in the report.
 
+## Phase 8: View API Audit
+
+**This phase is mandatory and blocks device testing.** Its purpose is to guarantee that no deprecated View-system API leaked into the new Compose code.
+
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md:180]: Duplicate of the ordering fix — addressed in aaefd2e.

File: plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md [hunk 5]
```diff
@@ -183,6 +183,101 @@ Brief the `manual-tester` agent with:
...
+Scan for any of these in migrated files:
+
+**Imports**
+- `import android.view.*`
+- `import android.widget.*`
+- `import android.app.Activity` (as base class)
+- `import android.app.Fragment` / `import androidx.fragment.app.Fragment`
+- `import android.databinding.*` / `import androidx.databinding.*`
+- `import android.viewbinding.*` / `import androidx.viewbinding.*`
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/migration-and-verify.md]: Fixed in aaefd2e. Removed android.viewbinding.* (nonexistent); the entry now reads androidx.viewbinding.* as suggested.

File: plugins/developer-workflow/skills/migrate-to-compose/references/custom-view-migration.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
+# Custom View Migration Strategies
+
+## When this applies
+
+Any project-owned custom View or ViewGroup used on the screen being migrated. Third-party Views (MapView, WebView, PlayerView, ExoPlayer) are exempt — wrap them in `AndroidView`.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/migrate-to-compose/references/custom-view-migration.md:5]: Not addressing in this PR to keep it focused on the Phase 8 audit gate. The ExoPlayer/PlayerView wording is a separate concern in a different file.
